cli/fathom_web/commands/vectorize.py

```diff
@@ -0,0 +1,312 @@
+    print('Done')
+    return firefox, firefox.capabilities['moz:processID'], firefox.service.process.pid
+
+
+def run_vectorizer(firefox, sample_filenames):
+    """
+    Set up the vectorizer and run it, creating the vectors file.
+
+    Navigate to the vectorizer page of FathomFox, paste the sample filenames
+    into the text area and hit the vectorize button.
+
+    We monitor the status text area for errors and to see how many samples
+    have been vectorized, so we know when the vectorizer has stopped running.
+    """
+    print('Configuring Vectorizer...', end='', flush=True)
+    # Navigate to the vectorizer page
+    fathom_fox_uuid = get_fathom_fox_uuid(firefox)
+    firefox.get(f'moz-extension://{fathom_fox_uuid}/pages/vector.html')
+
+    pages_text_area = firefox.find_element_by_id('pages')
+    pages_text_area.send_keys(sample_filenames)
+
+    downloads_dir = pathlib.Path(firefox.profile.default_preferences['browser.download.dir'])
+    vector_files_before = set(downloads_dir.glob('vector*.json'))
+    number_of_samples = len(sample_filenames.splitlines())
+    status_box = firefox.find_element_by_id('status')
+    vectorize_button = firefox.find_element_by_id('freeze')
+    completed_samples = 0
+    print('Done')
+
+    with progressbar(length=number_of_samples, label='Running Vectorizer...') as bar:
+        vectorize_button.click()
+        while completed_samples < number_of_samples:
+            if 'failed:' in status_box.text:
+                error = extract_error_from(status_box.text)
+                print(f'Vectorization failed with error:\n{error}')
+                break
+            now_completed_samples = len([line for line in status_box.text.splitlines() if line.endswith(': vectorized')])
+            bar.update(now_completed_samples - completed_samples)
+            completed_samples = now_completed_samples
+
+    new_file = look_for_new_vector_file(downloads_dir, vector_files_before)
+    print(f'Vectors saved to {str(new_file)}')
+    return firefox
+
+
+def get_fathom_fox_uuid(firefox):
+    """
+    Try to get the internal UUID for FathomFox from `prefs.js`.
+
+    We use a loop to try multiple times because the `prefs.js` file needs a
+    little time to update before the fathom addon information appears. Five
+    seconds seems adequate since one second has always worked for me (Daniel).
+    """
+    for _ in range(5):
+        prefs = (pathlib.Path(firefox.capabilities.get('moz:profile')) / 'prefs.js').read_text().split(';')
+        uuids = next((line for line in prefs if 'extensions.webextensions.uuids' in line)).split(',')
+        try:
+            fathom_fox_uuid = next((line for line in uuids if 'fathomfox@mozilla.com' in line)).split('\\"')[3]
+            return fathom_fox_uuid
+        except StopIteration:
+            time.sleep(1)
+    else:
+        raise GracefulError('Could not find UUID for FathomFox. No entry in the pres.js file.')
```

Comment:
I did think about generalizing this method when I wound up using it the second time, so I would support that.

Reply:
I don't see that this has been moved to a utilities function. Not blocking, but just want to remind you about it in case you want to do that before merging.

Reply:
I have moved it. It feels only okay to me. Let me know what you think about it.